client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java

```diff
@@ -412,12 +419,14 @@ public void run() {
                 }
                 
                 delayTime = serviceObj.getCacheMillis();
-                
+                failCount = 0;
             } catch (Throwable e) {
+                failCount++;
                 NAMING_LOGGER.warn("[NA] failed to update serviceName: " + serviceName, e);
             } finally {
                 if (delayTime > 0) {
-                    executor.schedule(this, delayTime, TimeUnit.MILLISECONDS);
+                    executor.schedule(this, Math.min(delayTime << failCount, DEFAULT_DELAY * 60),
```

Comment:
The implementation may cause `delayTime << failCount` become negative number when failCount=54.

Reply:
now, the failCunt cat't limit 6.